src/main/java/com/epam/esm/persistence/jdbc/gift/JdbcTemplateGiftDaoImpl.java

```diff
@@ -0,0 +1,196 @@
+package com.epam.esm.persistence.jdbc.gift;
+
+import com.epam.esm.persistence.dao.GiftCertificateDao;
+import com.epam.esm.persistence.util.search.QueryOrder;
+import com.epam.esm.persistence.util.search.GiftSearchFilter;
+import com.epam.esm.persistence.util.mapper.GiftMapper;
+import com.epam.esm.util.validation.BeforeGiftValidation;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.support.GeneratedKeyHolder;
+import org.springframework.jdbc.support.KeyHolder;
+import org.springframework.stereotype.Repository;
+
+import java.sql.Date;
+import java.sql.PreparedStatement;
+import java.sql.Statement;
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+
+@Repository
+public class JdbcTemplateGiftDaoImpl implements JdbcTemplateGiftDao {
```

Comment:
Use JdbcTemplateGiftCertificateDao name for DAO (JdbcTemplate already specifies that this is an implementation).